### ISSUE TITLE
Defense in Depth - Prevent attacker from becoming root within the container

### ISSUE THREAD

**Issue by apyrgio** [freedomofpress/dangerzone#224]
Parent issue: https://github.com/freedomofpress/dangerzone/issues/221

Once an attacker has found a flaw that allows them to take control of a process (see #222), they will probably attempt to become root within the container. Currently, this is not easy for two reasons:

1. The user within the container is a regular user (`dangerzone`, UID 1000).
2. We use `--security-opt=no-new-privileges`, so the kernel does not allow a process to elevate their privileges.

However, it may be worth ensureingthat it's not easy to become root in the container, even if this safeguard does not manage to work. For an extra layer of security, we could follow the latest CIS Docker advisories ([official link](https://www.cisecurity.org/benchmark/docker), [mirror](https://edu.anarcho-copy.org/GNU%20Linux%20-%20Unix-Like/Docker/CIS_Docker_Benchmark_v1.3.1_PDF.pdf)). 

Because the advisories are numerous, there are tools that check if a container image/runtime complies with the advisories:

* https://github.com/containers/podman-security-bench
* https://github.com/docker/docker-bench-security

I have already ran these two tools on Windows and Linux, and the noteworthy sections that were reported are the following:

* 4.3 - Ensure that unnecessary packages are not installed in the container
* 4.8 - Ensure setuid and setgid permissions are removed
  - The flagged binary was `sudo`, which is not used and can be removed.
* 5.12 - Ensure that the container's root filesystem is mounted as read only

We can easily placate section 4.8 by removing `sudo`, and section 5.12 by running the `podman` / `docker` command with the `--read-only` flag. The 4.3 section may require a bit more digging.

**Comment by deeplow** [comment]
4.3 can be considered part of the scope of https://github.com/freedomofpress/dangerzone/issues/232.

**Comment by apyrgio** [comment]
I agree and I updated the issue description. This issue should not close as long as #232 is open.

**Comment by apyrgio** [comment]
With the integration of Dangerzone with gVisor, we can now close this issue. Some requirements are not yet satisfied (such as the removal of the `sudo` binary), but due to the fact that we run the code in a nested gVisor container, we have gone above and beyond reducing the attacker's privileges.
